docs/propose_changelog.py

```diff
@@ -0,0 +1,186 @@
+        elif "plugin" in pr_labels_names:
+            bugs["plugin"].append(i.title + " #" + str(pr_nr))
+        elif "tools" in pr_labels_names:
+            bugs["tools"].append(i.title + " #" + str(pr_nr))
+        else:
+            bugs["other"].append(i.title + " #" + str(pr_nr))
+        continue
+    if "feature" in pr_labels_names:
+        if "core" in pr_labels_names:
+            features["core"].append(i.title + " #" + str(pr_nr))
+        elif "libPMacc" in pr_labels_names:
+            features["pmacc"].append(i.title + " #" + str(pr_nr))
+        elif "plugin" in pr_labels_names:
+            features["plugin"].append(i.title + " #" + str(pr_nr))
+        elif "tools" in pr_labels_names:
+            features["tools"].append(i.title + " #" + str(pr_nr))
+        else:
+            features["other"].append(i.title + " #" + str(pr_nr))
+        continue
+    if "refactoring" in pr_labels_names:
+        if "core" in pr_labels_names:
+            refactoring["core"].append(i.title + " #" + str(pr_nr))
+        elif "libPMacc" in pr_labels_names:
+            refactoring["pmacc"].append(i.title + " #" + str(pr_nr))
+        elif "plugin" in pr_labels_names:
+            refactoring["plugin"].append(i.title + " #" + str(pr_nr))
+        elif "tools" in pr_labels_names:
+            refactoring["tools"].append(i.title + " #" + str(pr_nr))
+        else:
+            refactoring["other"].append(i.title + " #" + str(pr_nr))
+        continue
+    if "documentation" in pr_labels_names:
+        misc["docs"].append(i.title + " #" + str(pr_nr))
+        continue
+    misc["other"].append(i.title + " #" + str(pr_nr))
+
+print("")
+print("**New Features:**")
+print(" - PIC:")
+for p in features["core"]:
+    print("   - " + p)
+print(" - libPMacc:")
+for p in features["pmacc"]:
+    print("   - " + p)
+print(" - plugins:")
+for p in features["plugin"]:
+    print("   - " + p)
+print(" - tools:")
+for p in features["tools"]:
+    print("   - " + p)
+for p in features["other"]:
+    print(" - " + p)
+
+print("")
+print("**Bug Fixes:**")
+print(" - PIC:")
+for p in bugs["core"]:
+    print("   - " + p)
+print(" - libPMacc:")
+for p in bugs["pmacc"]:
+    print("   - " + p)
+print(" - plugins:")
+for p in bugs["plugin"]:
+    print("   - " + p)
+print(" - tools:")
+for p in bugs["tools"]:
+    print("   - " + p)
+for p in bugs["other"]:
+    print(" - " + p)
+
+print("")
+print("**Misc:**")
+print(" - refactoring:")
+print("   - PIC:")
+for p in refactoring["core"]:
+    print("     - " + p)
+print("   - libPMacc:")
+for p in refactoring["pmacc"]:
+    print("     - " + p)
+print("   - plugins:")
+for p in refactoring["plugin"]:
+    print("     - " + p)
+print("   - tools:")
+for p in refactoring["tools"]:
+    print("     - " + p)
+for p in refactoring["other"]:
+    print("   - " + p)
+print(" - documentation:")
+for p in misc["docs"]:
+    print("   - " + p)
+for p in misc["other"]:
+    print(" - " + p)
```

Comment:
NEARLY REQUIRED:

Great Job with this automated tool - you could shorten this repetitive structure by using decorators or even simply a function to avoid this massive code duplication: 

```python
# function
def print_fkt(data, key):
    for p in data[kex]:
      print(" - " + p)

# more missing ...

print("")
print("**Misc:**")
print(" - refactoring:")
print("   - PIC:")
print_fkt(refactoring, "core")
print("   - libPMacc:")
print_fkt(refactoring, "pmacc")
# ... still more missing
```